packages/backend-common/src/scm/git.ts

```diff
@@ -42,19 +73,21 @@ instead of Basic Auth (e.g., Bitbucket Server).
  *
  * @public
  */
+
 export class Git {
   private readonly headers: {
     [x: string]: string;
   };
 
   private constructor(
     private readonly config: {
-      username?: string;
-      password?: string;
+      onAuth?: AuthCallback;
       token?: string;
       logger?: LoggerService;
     },
   ) {
+    this.onAuth = config.onAuth ?? noAuth;
```

Comment:
Shouldn't this default to the older auth instead to keep backwards compatibility?

Reply:
I made `onAuth` required when using the `git.fromAuth(...)` static factory. I made it optional because `username` and `password` were optional, but on second thought, if you decide to invoke `git.fromAuth(...)` with `{ onAuth: ...}`, then you should also pass in the `callback`. I think the `isAuthCallbackOptions` type predicate already skipped options that did not specify `onAuth` anyway.

As a side-effect, if you leave the options empty, the same behaviour as before is used, creating a callback that will return `{ username: undefined, password: undefined }`.